From d57d3e9812159ff6e8df788049c8a903e1517b7b Mon Sep 17 00:00:00 2001
From: Rene Engelhard <rene@rene-engelhard.de>
Date: Sat, 9 Aug 2025 19:05:15 +0200
Subject: [PATCH] add EUR for Bulgaria/Lew

Bulgaria will join Eurozone 2026-01-01 at which time around the default
needs to be switched

see https://www.ecb.europa.eu/press/pr/date/2025/html/ecb.pr250708~b9676a9fa8.de.html for the conversion rate

This is the < 26.2 version including the euro wizard (cf.
b1a2f727ca99ecd3402d4b051b99cbfd24266e59)

Change-Id: I840ddda4308549703c53f998812848888dd830f6
Reviewed-on: https://gerrit.libreoffice.org/c/core/+/189279
Reviewed-by: Eike Rathke <erack@redhat.com>
Tested-by: Jenkins
---
 i18npool/source/localedata/data/bg_BG.xml        |  8 ++++++++
 .../registry/data/org/openoffice/Office/Calc.xcu | 11 +++++++++++
 sc/source/core/tool/interpr2.cxx                 |  3 ++-
 wizards/source/euro/Init.xba                     | 16 ++++++++++++++++
 .../source/resources/resources_en_US.properties  |  1 +
 5 files changed, 38 insertions(+), 1 deletion(-)

diff --git a/i18npool/source/localedata/data/bg_BG.xml b/i18npool/source/localedata/data/bg_BG.xml
index 55c63049089f..ae295a38c7b9 100644
--- a/i18npool/source/localedata/data/bg_BG.xml
+++ b/i18npool/source/localedata/data/bg_BG.xml
@@ -366,6 +366,14 @@
       <CurrencyName>BGL</CurrencyName>
       <DecimalPlaces>2</DecimalPlaces>
     </Currency>
+    <!-- TODO: switch defaults before 2026-01-01 -->
+    <Currency default="false" usedInCompatibleFormatCodes="false">
+      <CurrencyID>EUR</CurrencyID>
+      <CurrencySymbol>€</CurrencySymbol>
+      <BankSymbol>EUR</BankSymbol>
+      <CurrencyName>евро</CurrencyName>
+      <DecimalPlaces>2</DecimalPlaces>
+    </Currency>
   </LC_CURRENCY>
   <LC_TRANSLITERATION ref="en_US"/>
   <LC_MISC>
diff --git a/officecfg/registry/data/org/openoffice/Office/Calc.xcu b/officecfg/registry/data/org/openoffice/Office/Calc.xcu
index eda60fe6c434..2b1e0e2bf5a3 100644
--- a/officecfg/registry/data/org/openoffice/Office/Calc.xcu
+++ b/officecfg/registry/data/org/openoffice/Office/Calc.xcu
@@ -239,6 +239,17 @@
         <value>7.53450</value>
       </prop>
     </node>
+    <node oor:name="CR21" oor:op="replace">
+      <prop oor:name="FromUnit">
+        <value>EUR</value>
+      </prop>
+      <prop oor:name="ToUnit">
+        <value>BLN</value>
+      </prop>
+      <prop oor:name="Factor">
+        <value>1.95583</value>
+      </prop>
+    </node>
   </node>
   <node oor:name="Calculate">
     <node oor:name="Other">
diff --git a/sc/source/core/tool/interpr2.cxx b/sc/source/core/tool/interpr2.cxx
index a2edd03c9188..c71bbc39d88d 100644
--- a/sc/source/core/tool/interpr2.cxx
+++ b/sc/source/core/tool/interpr2.cxx
@@ -3299,7 +3299,8 @@ static bool lclConvertMoney( std::u16string_view aSearchUnit, double& rfRate, in
         { "EEK", 15.6466,  2 },
         { "LVL", 0.702804, 2 },
         { "LTL", 3.45280,  2 },
-        { "HRK", 7.53450,  2 }
+        { "HRK", 7.53450,  2 },
+        { "BLN", 1.95583,  2 }
     };
 
     for (const auto & i : aConvertTable)
diff --git a/wizards/source/euro/Init.xba b/wizards/source/euro/Init.xba
index 250b0e5a7639..cd2c4c3e8016 100644
--- a/wizards/source/euro/Init.xba
+++ b/wizards/source/euro/Init.xba
@@ -90,6 +90,7 @@ Public sCurrESTONIAN as String
 Public sCurrLATVIAN as String
 Public sCurrLITHUANIAN as String
 Public sCurrCROATIAN as String
+Public sCurrBULGARIAN as String
 
 Public sPrgsRETRIEVAL as String
 Public sPrgsCONVERTING as String
@@ -216,6 +217,7 @@ Dim LocWorkPath as String
 		sCurrLATVIAN = GetResText(&quot;CURRENCIES_17&quot;)
 		sCurrLITHUANIAN = GetResText(&quot;CURRENCIES_18&quot;)
 		sCurrCROATIAN = GetResText(&quot;CURRENCIES_19&quot;)
+		sCurrBULGARIAN = GetResText(&quot;CURRENCIES_20&quot;)
 		.cmdCancel.Label =  sCANCEL
 		.cmdHelp.Label =  sHELP
 		.cmdBack.Label =  GetResText(&quot;STEP_ZERO_2&quot;)
@@ -400,6 +402,11 @@ Sub InitializeLanguages()
     LangIDValue(19,0,1) = &quot;HR&quot;
     LangIDValue(19,0,2) = &quot;-41A&quot;
 
+&apos; CURRENCIES_BULGARIAN
+    LangIDValue(20,0,0) = &quot;bg&quot;
+    LangIDValue(20,0,1) = &quot;BG&quot;
+    LangIDValue(20,0,2) = &quot;-402&quot;
+
 End Sub
 
 
@@ -588,6 +595,15 @@ Dim i as Integer
 	CurrValue(19,4) = &quot;kn&quot;
 	CurrValue(19,5) = &quot;HRK&quot;
 
+	CurrValue(20,0) = sCurrBULGARIAN
+	&apos; real conversion rate
+	CurrValue(20,1) = 1.95583
+	&apos; rounded conversion rate
+	CurrValue(20,2) = 2
+	CurrValue(20,3) = &quot;лв.&quot;
+	CurrValue(20,4) = &quot;лв.&quot;
+	CurrValue(20,5) = &quot;BLN&quot;
+
 	i = -1
 	CurrSymbolList(0) = &quot;&quot;
 	CurrSymbolList(1) = &quot;&quot;
diff --git a/wizards/source/resources/resources_en_US.properties b/wizards/source/resources/resources_en_US.properties
index d75cb03bdf05..5ae6382ccafa 100644
--- a/wizards/source/resources/resources_en_US.properties
+++ b/wizards/source/resources/resources_en_US.properties
@@ -449,6 +449,7 @@ CURRENCIES_16=Estonian Kroon
 CURRENCIES_17=Latvian Lats
 CURRENCIES_18=Lithuanian Litas
 CURRENCIES_19=Croatian Kuna
+CURRENCIES_20=Bulgarian Lew
 STEP_LASTPAGE_0=Progress
 STEP_LASTPAGE_1=Retrieving the relevant documents...
 STEP_LASTPAGE_2=Converting the documents...
-- 
2.47.2

